### PR TITLE
Add JDBC setup documentation

### LINKED ISSUE #1806: Document how to connect DBs using vividus-plugin-db
Document how to connect popular DBs:
 - PostgreSQL
 - Azure SQL
 - MySQL
 - ...

The documentation should include:
- the references to driver implementations (gradle dependency)
- ~driver class names~ (they are optional actually)
- examples of JDBC URLs

### LINKED ISSUE #2977: [docs] Document data base connection via VIVIDUS
1. Document generic algorithm
2. Provide examples of configuration for: postgresql, mssql, mysql databases

### PR DESCRIPTION
**Pull request by Aliaksei-Andreyeu**
Closes #2977,  Closes #1806
Adds JDBC introduction and provides setup examples.

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 1]
```diff
@@ -1,19 +1,127 @@
...
+    - jdbc: - reference to JDBC;
+    - dbms: - Database Management System, e.g. `postgresql`;
+    - //databaseUrl - address to your database;
+    - :port - port number, e.g. 5432;
+    - /databaseName - name of your database;
+    - parameter - any additional parameters you need to pass;
++
+IMPORTANT: URL structure slightly changes depending on the database management system. Below you can find templates for different DBMS
++
+.Click to see URL templates
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:60]: let's move these templates to `Setup examples` so to keep information about particular DBMS-s grouped (also you can add a reference to `Setup examples` from this point)

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 2]
```diff
@@ -1,19 +1,127 @@
...
+db.connection.vividus.username=vividus
+db.connection.vividus.password=passwordValue
+----
+
+==== Microsoft SQL
+
+.environment.properties
+[source,properties,subs="attributes+"]
+----
+db.connection.vividus.url=jdbc:sqlserver://mssql.db.server\\mssql_instance;databaseName=my_database
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:111]:
```suggestion
db.connection.vividus.url=jdbc:sqlserver://mssql.db.server\mssql_instance;databaseName=my_database
```
 <- are doubled slashes needed?

Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:111]: also from my experience port is required almost for all cases, AFAIK it's `1433` by default

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 3]
```diff
@@ -1,19 +1,127 @@
...
+db.connection.vividus.username=vividus
+db.connection.vividus.password=passwordValue
+----
+
+==== PostgreSQL
+
+.environment.properties
+[source,properties,subs="attributes+"]
+----
+db.connection.vividus.url=jdbc:postgresql://postgresql.db.server:5430/my_database?ssl=true&loglevel=2
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:121]:
```suggestion
db.connection.vividus.url=jdbc:postgresql://postgresql.db.server:5432/my_database?ssl=true&loglevel=2
```
<- let's use common defaults, it'll help to align driver docs and our docs

File: docs/modules/user-guides/pages/use-in-memory-db-to-manage-test-data.adoc [hunk 4]
```diff
@@ -113,5 +113,5 @@ UPDATE test.students SET firstname='Mary' WHERE lastname='Miller';
 When I execute SQL query `
 SELECT * FROM test.students
 ` against `testh2db` and save result to scenario variable `h2-records`
-Then `${h2-records}` is equal to `[{firstname=Mary, lastname=Miller}, {firstname=Alex, lastname=Magnus}]`
+Then `${h2-records}` is equal to `[{FIRSTNAME=Mary, LASTNAME=Miller}, {FIRSTNAME=Alex, LASTNAME=Magnus}]`
```
Comment (valfirst) [inline docs/modules/user-guides/pages/use-in-memory-db-to-manage-test-data.adoc:116]: is it required change?

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 5]
```diff
@@ -1,19 +1,127 @@
...
+db.connection.vividus.username=vividus
+db.connection.vividus.password=passwordValue
+----
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:124]: could you please also add:
- CSV (and add a reference to our `Execute SQL against CSV` guide)
- H2 (and add a reference to our `Use in-memory database` guide)
?

**Review by valfirst — CHANGES_REQUESTED** [review]

**Comment by valfirst** [comment]
also I would add driver dependency examples for each of the listed DBMS to make it super-comfortable for users

File: docs/modules/user-guides/pages/use-in-memory-db-to-manage-test-data.adoc (hunk 4 above)
Comment (Aliaksei-Andreyeu) [reply docs/modules/user-guides/pages/use-in-memory-db-to-manage-test-data.adoc:116]: Not really related to the issue.
When I studied H2 and ran scenarios, one of them failed:
<details>
Screenshot 2022-07-27 at 15 16 59
</details>

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 6]
```diff
@@ -1,19 +1,179 @@
...
++
+.IDEA
+[source,shell]
+----
+./gradlew cleanIdea idea --refresh-dependencies
+----
+. Refresh IDE project (or restart IDE);
+. Copy provided properties;
+. Update URL, username and password according to your database.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:82]:
```suggestion
Here are some examples of setting different databases connection.
```
let's remove IDE configuration part, it's very generic and can be moved to some other part, e.g. `Getting Started`

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 7]
```diff
@@ -1,19 +1,179 @@
...
+
+==== MySQL
+
+.build.gradle
+[source,shell]
+----
+implementation(group: 'org.vividus', name: 'vividus-plugin-db', version: '0.4.14-SNAPSHOT')
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:89]: I'm doubtful we need to add this line to each example, from my point of view the approach is clearly described in `Installation` section

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 8]
```diff
@@ -1,19 +1,179 @@
...
+implementation(group: 'mysql', name: 'mysql-connector-java', version: '8.0.29')
+----
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:91]:
```
.URL template
[source,URL,subs="attributes+"]
----
jdbc:mysql://host:port/database?properties
----
```

Comment (valfirst) [reply docs/modules/plugins/pages/plugin-db.adoc:91]: my idea was to put URL templates in the corresponding DBMS sections

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**